common/hooks/useDebounce.ts

```diff
@@ -0,0 +1,19 @@
+import { useState, useEffect } from 'react';
+
+const useDebounce = <T>(value: T, delay: number) => {
+  const [debouncedValue, setDebouncedValue] = useState(value);
+
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setDebouncedValue(value);
+    }, delay);
```

Comment:
Nå er jeg litt usikker på om dette er funksjonaliteten vi faktisk ønsker?

Med dette vil du fra første bokstav du trykker på utsette når søket skal utføres. Så uansett hva jeg skriver vil den vente minimum 1,5 sekunder fra jeg er **ferdig** med å skrive til den faktisk utfører et søk?

Etter å ha prøvd litt ut i vercel-lenken virker det for meg enda tregere med denne metoden. Ikke ser det ut til at backenden blir noe raskere ved at det utføres færre kall heller?